core/src/main/java/com/linecorp/armeria/client/Bootstraps.java

```diff
@@ -131,4 +115,92 @@ private static int toIndex(boolean webSocket) {
     private static int toIndex(SerializationFormat serializationFormat) {
         return toIndex(serializationFormat == SerializationFormat.WS);
     }
+
+    /**
+     * Returns a {@link Bootstrap} corresponding to the specified {@link SocketAddress}
+     * {@link SessionProtocol} and {@link SerializationFormat}.
+     */
+    Bootstrap getOrCreate(SocketAddress remoteAddress, SessionProtocol desiredProtocol,
+                          SerializationFormat serializationFormat) {
+        if (!httpAndHttpsValues().contains(desiredProtocol)) {
+            throw new IllegalArgumentException("Unsupported session protocol: " + desiredProtocol);
+        }
+
+        final boolean isDomainSocket = remoteAddress instanceof DomainSocketAddress;
+        if (isDomainSocket && unixBaseBootstrap == null) {
+            throw new IllegalArgumentException("Domain sockets are not supported by " +
+                                               eventLoop.getClass().getName());
+        }
+
+        if (sslContextFactory == null || !desiredProtocol.isTls()) {
+            return select(isDomainSocket, desiredProtocol, serializationFormat);
+        }
+
+        final Bootstrap baseBootstrap = isDomainSocket ? unixBaseBootstrap : inetBaseBootstrap;
+        assert baseBootstrap != null;
+        return newBootstrap(baseBootstrap, remoteAddress, desiredProtocol, serializationFormat);
```

Comment:
So, we should create bootstrap every time when connecting if `sslContextFactory` is set?

Reply:
Right. Because we need a new `ChannelInitializer` for a different `SslContext`.

Reply:
If so, I'm a bit worried about the performance. 🤔 
Maybe @trustin can give some idea.

Reply:
When I saw `Bootstrap`, it was a simple class and did not refer to heavy objects.